root/static/scripts/guess-case/modes.js

```diff
@@ -282,3 +291,20 @@ exports.Sentence = assign({}, DefaultMode, {
     {url: {href: 'https://musicbrainz.org/doc/Style/Language', target: '_blank'}}
   ),
 });
+
+exports.Turkish = assign({}, DefaultMode, {
+  description: l(
+    'This mode handles the Turkish capitalisation of \'i\' (\'İ\') and \'ı\' '+
+    '(\'I\'). Some words may need to be manually corrected '+
+    'according to the {url|Turkish language guidelines}. ',
+    {url: {href: 'https://musicbrainz.org/doc/Style/Language/Turkish', target: '_blank'}}
+  ),
+
+  isSentenceCaps() {
+    return false;
+  },
+
+  toUpperCase:turkishUpperCase,
+
+  toLowerCase:turkishLowerCase,
```

Comment:
There is one tiny style issue remaining, I would like to have a space after each of the colons.

But it’s now ripe for merging, I guess.